### PR TITLE
Coverage Run unit tests first before docker containers are set up

### LINKED ISSUE #1051: [feat] Run unit test in CI on pull request
### Feature Request / Improvement

It's easy to forget to add the "integration test mark" for integration tests. For example, [`test_table_scan_default_to_large_types`](https://github.com/apache/iceberg-python/blob/f05b1aedee8451d981188adf68be5e8b360a9ca1/tests/integration/test_reads.py#L755-L757) is currently missing the integration test mark since it depends on the hive catalog. 

We can solve this by running `make test` in CI. `make test` will fail for the above since the hive catalog is not available.

### PR DESCRIPTION
**Pull request by Minfante377**
This PR implements:
- Run unit tests as part of the CI
- Add a missing integration mark to one of the integration tests

Should close #1051

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -67,11 +73,13 @@ test-coverage:
 	sleep 10
 	docker compose -f dev/docker-compose-integration.yml cp ./dev/provision.py spark-iceberg:/opt/spark/provision.py
 	docker compose -f dev/docker-compose-integration.yml exec -T spark-iceberg ipython ./provision.py
-	poetry run coverage run --source=pyiceberg/ -m pytest tests/ ${PYTEST_ARGS}
+	poetry run coverage run --source=pyiceberg/ -m pytest tests/ -m integration ${PYTEST_ARGS}
```
Comment (sungwy) [inline Makefile:76]: If we set the `--data-file` argument to a specific value for the unit tests (`.coverage.unit`), and then a different one for the integration test (`.coverage.integration`), then we could `coverage combine` these data files and then produce the `coverage report` together.

**Review by sungwy — APPROVED** [review]
Thank you @Minfante377 for this PR, and @youssef-itanii as well!

File: Makefile [hunk 2]
```diff
@@ -67,7 +70,10 @@ test-coverage:
 	sleep 10
 	docker compose -f dev/docker-compose-integration.yml cp ./dev/provision.py spark-iceberg:/opt/spark/provision.py
 	docker compose -f dev/docker-compose-integration.yml exec -T spark-iceberg ipython ./provision.py
-	poetry run coverage run --source=pyiceberg/ -m pytest tests/ ${PYTEST_ARGS}
+	poetry run coverage run --data-file=.coverage.integration --source=pyiceberg/ -m pytest tests/ -m integration ${PYTEST_ARGS}
```
Comment (kevinjqliu) [inline Makefile:73]: nit: to match the other command, also added `-v` to match `test-integration`
```suggestion
	poetry run coverage run --source=pyiceberg/ --data-file=.coverage.integration -m pytest tests/ -v -m integration ${PYTEST_ARGS}
```

File: tests/integration/test_reads.py [hunk 3]
```diff
@@ -753,6 +753,7 @@ def test_configure_row_group_batch_size(session_catalog: Catalog) -> None:
     assert len(batches) == entries
 
 
+@pytest.mark.integration
```
Comment (kevinjqliu) [inline tests/integration/test_reads.py:756]: Does CI fail if we don't add this line? It's what we're testing for in #1051

**Review by kevinjqliu — COMMENTED** [review]
Thanks for the contribution! Added a comment about testing this method and some nits

File: Makefile (hunk 2 above)
Comment (Minfante377) [reply Makefile]: Done

File: tests/integration/test_reads.py (hunk 3 above)
Comment (Minfante377) [reply tests/integration/test_reads.py:756]: It fails indeed. If I don't add that marker the `test-coverage-unit` command will try to run that test without initializing any docker container

**Review by kevinjqliu — APPROVED** [review]
LGTM!

Verified that unit test fails 
https://github.com/apache/iceberg-python/pull/1060
https://github.com/apache/iceberg-python/actions/runs/10380009878/job/28739139875?pr=1060
```
=========================== short test summary info ============================
FAILED tests/integration/test_reads.py::test_table_scan_default_to_large_types[session_catalog_hive] - thrift.transport.TTransport.TTransportException: Could not connect to any of [('::1', 9083, 0, 0), ('127.0.0.1', 9083)]
ERROR tests/integration/test_reads.py::test_table_scan_default_to_large_types[session_catalog] - requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8181): Max retries exceeded with url: /v1/config (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f2b1351e0d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
=== 1 failed, 2781 passed, 834 deselected, 1113 warnings, 1 error in 54.83s ====
```

**Pull request merged**